src/ess/spectroscopy/indirect/io.py

```diff
@@ -0,0 +1,191 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+from __future__ import annotations
+
+import h5py
+import numpy
+import scippnexus
+from scipp import DataArray
+
+from ..types import NormWavelengthEvents, NXspeFileName, NXspeFileNames
+
+
+def to_nxspe(events: NormWavelengthEvents, base: NXspeFileName) -> NXspeFileNames:
+    """Take events, which have been binned in incident wavelength and have monitor
+    counts per bin, and output one NXspe file per setting
+
+    Parameters
+    ----------
+    events: scipp.DataArray
+        The events binned in (setting, event_id, incident_wavelength) with
+        at least 'monitor', 'a3', and 'theta' bin coordinates.
+        The events and the bins must also have an 'incident_wavelength' coordinate.
+    base: str | Path
+        The filename base used to produce each NXspe filename.
+
+    Returns
+    -------
+    :
+        The list of filenames containing the NXspe data. If there are N settings in the
+        input DataArray, N filenames are returned. The names are of the form
+            {base}_{i+1:0{ceil(log10(N+1))}d}.nxspe
+    """
+    from pathlib import Path
+
+    from tqdm import tqdm
```

Comment:
Please remove. I get that it is nice to show progress. But we don't do this in library code because we can't predict how it will eventually be used.

How long does it take to write the files?

Reply:
About a minute for 180 files. 
`100%|██████████████████████████████████| 180/180 [00:57<00:00,  3.15it/s]`

Which I think is too long for a command to give no feedback. What if the progress display was controlled by a keyword argument?

Reply:
Progress bar removed.